src/holodeck/environments.py

```diff
@@ -425,6 +425,60 @@ def add_agent(self, agent_def, is_main_agent=False):
         if is_main_agent:
             self._agent = self.agents[agent_def.name]
 
+    def spawn_prop(self, prop_type, location=None, rotation=None, scale=1, 
+                    sim_physics=False, material="", name=""):
+        """Spawns a basic prop object in the world like a box or sphere.
+
+        Args:
+            prop_type (:obj:`string`):
+                The type of prop to spawn. Can be ``box``, ``sphere``, ``cylinder``, or ``cone``.
+
+            location (:obj:`list` of :obj:`float`):
+                The ``[x, y, z]`` location of the prop.
+
+            rotation (:obj:`list` of :obj:`float`):
+                The ``[roll, pitch, yaw]`` rotation of the prop.
+
+            scale (:obj:`list` of :obj:`float`) or (:obj:`float`):
+                The ``[x, y, z]`` scalars to the prop size, where the default size is 1 meter.
+                If given a single float value, then every dimension will be scaled to that value.
+
+            sim_physics (:obj:`boolean`):
+                Whether the object is mobile and is affected by gravity and collisions.
+
+            material (:obj:`string`):
+                The type of material (texture) to apply to the prop. Can be ``white``, ``gold``,
+                ``cobblestone``, ``brick``, ``wood``, ``grass``, ``steel``, or ``black``. If left
+                empty, the prop will have the default unreal material.
+
+            name (:obj:`string`):
+                The name to apply to the prop. Useful for task references, like the
+                :ref:`location-task`.
+        """
+        location = [0, 0, 0] if location is None else location
+        rotation = [0, 0, 0] if rotation is None else rotation
+        # if the given scale is an single value, then scale every dimension to that value
+        if not isinstance(scale, list):
+            scale = [scale, scale, scale]
+        sim_physics = 1 if sim_physics else 0
+
+        prop_type = prop_type.lower()
+        material = material.lower()
+
+        available_props = ["box", "sphere", "cylinder", "cone"]
```

Comment:
add teapot

Reply:
;)

Reply:
No - you have to call send world command directly. That way it is super secret.

Reply:
I mean -- what teapot?